pkg/kclient/ingress.go

```diff
@@ -3,15 +3,41 @@ package kclient
 import (
 	"context"
 	"fmt"
-
+	"github.com/openshift/odo/pkg/unions"
 	"github.com/pkg/errors"
 
 	extensionsv1 "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CreateIngress creates an ingress object for the given service and with the given labels
-func (c *Client) CreateIngress(ingress extensionsv1.Ingress) (*extensionsv1.Ingress, error) {
+func (c *Client) CreateIngress(ingress unions.KubernetesIngress) (*unions.KubernetesIngress, error) {
+	var err error
+	if !ingress.IsGenerated() {
+		return nil, fmt.Errorf("please pass a generated ingress")
+	}
+	created := false
+	kii := unions.NewEmptyKubernetesIngress()
+	if c.isNetworkingV1IngressSupported {
+		kii.NetworkingV1Ingress, err = c.KubeClient.NetworkingV1().Ingresses(c.Namespace).Create(context.TODO(), ingress.NetworkingV1Ingress, metav1.CreateOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create networking v1 ingress %w", err)
+		}
+		created = true
+	} else if c.isExtensionV1Beta1IngressSupported {
+		kii.ExtensionV1Beta1Ingress, err = c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).Create(context.TODO(), ingress.ExtensionV1Beta1Ingress, metav1.CreateOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create extension v1 beta 1 ingress %w", err)
+		}
+		created = true
+	}
+	if !created {
+		return nil, fmt.Errorf("failed to create ingress as none of the options are supported")
+	}
+	return kii, nil
+}
+
+// CreateExtensionV1Beta1Ingress creates an ingress object for the given service and with the given labels
```

Comment:
I'm not sure the documentation is still accurate. I cannot see any labels

Reply:
Updated

Reply:
This is the old function by the way